src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt

```diff
@@ -65,13 +67,29 @@ class PersonService(
     if (isUpdateEvent(event)) {
       trackEvent(CPR_UPDATE_RECORD_DOES_NOT_EXIST, person)
     }
+    val personKey: PersonKeyEntity? = when {
+      person.isAboveMatchScoreThreshold -> getPersonKey(person)
+      else -> handleLowSelfMatchScore(person)
+    }
+    createPersonEntity(person, personKey)
+    trackEvent(TelemetryEventType.CPR_RECORD_CREATED, person)
+  }
+
+  private fun getPersonKey(person: Person): PersonKeyEntity {
     val personEntity = searchByAllSourceSystemsAndHasUuid(person)
-    val personKey = when {
+    return when {
       personEntity == null -> createPersonKey(person)
       else -> retrievePersonKey(person, personEntity)
     }
-    createPersonEntity(person, personKey)
-    trackEvent(TelemetryEventType.CPR_RECORD_CREATED, person)
+  }
+
+  private fun handleLowSelfMatchScore(person: Person): PersonKeyEntity? {
+    trackEvent(
+      TelemetryEventType.CPR_LOW_SELF_SCORE_NOT_CREATING_UUID,
+      person,
+      mapOf(EventKeys.PROBABILITY_SCORE to person.selfMatchScore.toString()),
+    )
+    return null
```

Comment:
I can see why this is returning null but again it is a bit confusing - it is the absence of an existing person

Reply:
Improve readability 👍